llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -8291,6 +8291,17 @@ static std::optional<bool> isImpliedCondICmps(const ICmpInst *LHS,
   if (areMatchingOperands(L0, L1, R0, R1, AreSwappedOps))
     return isImpliedCondMatchingOperands(LPred, RPred, AreSwappedOps);
 
+  if (LPred == ICmpInst::ICMP_SGT) {
```

Comment:
Can we handle more predicates?

Reply:
hi @dtcxzyw
Because different predicates have some different logics to address,
I think it is fine to keep this PR to fix the issue in the comment link,
and I'll try to expand some others implemented separately.

Reply:
They don't seem all that different (for `sge`, `uge`, and `ugt` that is).
https://alive2.llvm.org/ce/z/aNpE5W

You only need the `C <= 0` for signed predicates.
For signed you need `nsw`. For unsigned you need `nuw`.

Reply:
Thanks, I'll extend this